core/src/main/java/io/kestra/core/runners/NamespaceFilesService.java

```diff
@@ -0,0 +1,127 @@
+package io.kestra.core.runners;
+
+import io.kestra.core.models.tasks.NamespaceFiles;
+import io.kestra.core.serializers.JacksonMapper;
+import io.kestra.core.storages.FileAttributes;
+import io.kestra.core.storages.StorageInterface;
+import io.micronaut.core.annotation.Nullable;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.kestra.core.utils.Rethrow.*;
+
+@Singleton
+@Slf4j
+public class NamespaceFilesService {
+    @Inject
+    private StorageInterface storageInterface;
+
+    public List<URI> inject(RunContext runContext, String tenantId, String namespace, Path basePath, Object namespaceFiles) throws Exception {
```

Comment:
```suggestion
    public List<URI> list(RunContext runContext, String tenantId, String namespace, Path basePath, Object namespaceFiles) throws Exception {
```

It didn't really inject the files but list the files to be injected by the task that uses it so the name is kind of misleading.

Reply:
this method copy the files on the directory and return the list of copied files, inject is better than list